capsules/src/usb/cdc.rs

```diff
@@ -0,0 +1,496 @@
+    /// well, if we have a buffer then we are actively doing a receive.
+    rx_buffer: TakeCell<'static, [u8]>,
+    /// How many bytes the client wants us to receive.
+    rx_len: Cell<usize>,
+    /// How many bytes we have received so far.
+    rx_offset: Cell<usize>,
+    /// The RX client to use when RX data is received.
+    rx_client: OptionalCell<&'a dyn uart::ReceiveClient>,
+}
+
+impl<'a, U: hil::usb::UsbController<'a>> CdcAcm<'a, U> {
+    pub fn new(controller: &'a U, max_ctrl_packet_size: u8) -> Self {
+        let interfaces: &mut [InterfaceDescriptor] = &mut [
+            InterfaceDescriptor {
+                interface_number: 0,
+                interface_class: 0x02,    // CDC communication
+                interface_subclass: 0x02, // abstract control model (ACM)
+                interface_protocol: 0x01, // V.25ter (AT commands)
+                ..InterfaceDescriptor::default()
+            },
+            InterfaceDescriptor {
+                interface_number: 1,
+                interface_class: 0x0a,    // CDC data
+                interface_subclass: 0x00, // none
+                interface_protocol: 0x00, // none
+                ..InterfaceDescriptor::default()
+            },
+        ];
+
+        let cdc_descriptors: &mut [CdcInterfaceDescriptor] = &mut [
+            CdcInterfaceDescriptor {
+                subtype: descriptors::CdcInterfaceDescriptorSubType::Header,
+                field1: 0x10, // CDC
+                field2: 0x11, // CDC
+            },
+            CdcInterfaceDescriptor {
+                subtype: descriptors::CdcInterfaceDescriptorSubType::CallManagement,
+                field1: 0x00, // Capabilities
+                field2: 0x01, // Data interface 1
+            },
+            CdcInterfaceDescriptor {
+                subtype: descriptors::CdcInterfaceDescriptorSubType::AbstractControlManagement,
+                field1: 0x06, // Capabilities
+                field2: 0x00, // unused
+            },
+            CdcInterfaceDescriptor {
+                subtype: descriptors::CdcInterfaceDescriptorSubType::Union,
+                field1: 0x00, // Interface 0
+                field2: 0x01, // Interface 1
+            },
+        ];
+
+        let endpoints: &[&[EndpointDescriptor]] = &[
+            &[EndpointDescriptor {
+                endpoint_address: EndpointAddress::new_const(4, TransferDirection::DeviceToHost),
+                transfer_type: TransferType::Interrupt,
+                max_packet_size: 8,
+                interval: 100,
+            }],
+            &[
+                EndpointDescriptor {
+                    endpoint_address: EndpointAddress::new_const(
+                        2,
+                        TransferDirection::DeviceToHost,
+                    ),
+                    transfer_type: TransferType::Bulk,
+                    max_packet_size: 64,
+                    interval: 100,
```

Comment:
The interval for bulk endpoints is ignored, so setting this to zero would be preferable so people don't puzzle about its meaning. https://www.beyondlogic.org/usbnutshell/usb5.shtml#EndpointDescriptors

```suggestion
                    interval: 0,
```